### PR TITLE
Add some comment to clarify 5b4aa213

### PR DESCRIPTION
**Pull request by ryoqun**
well, it took a while to reason this odd port handling.

to unblock: https://github.com/solana-labs/solana/pull/12245

### CODE REVIEW

File: src/connection.js [hunk 1]
```diff
@@ -1462,6 +1462,12 @@ export class Connection {
 
     url.protocol = url.protocol === 'https:' ? 'wss:' : 'ws:';
     url.host = '';
+    // Only shift the port by +1 as a convention for ws(s) only if given endpoint
+    // is explictly specifying the endpoint port (HTTP-based RPC), assuming
+    // we're directly trying to connect to solana-validator's ws listening port.
+    // When the endpoint omits the port, we're connecting to the protocol
+    // default ports: http(80) or https(443) and it's assumed we're behind a reverse
+    // proxy which manages WebSocket upgrade and backend port redirection.
```
Comment (ryoqun) [inline src/connection.js:1470]: @mvines could you check my understanding?

Comment (ryoqun) [reply src/connection.js:1470]: 5b4aa213

Comment (mvines) [reply src/connection.js:1470]: Yep, exactly right.  Thanks for adding this helpful detail

**Review by mvines — APPROVED** [review]

**Comment by mvines** [comment]
Oops.  I didn't realize this PR was directly into solana-web3.js.  This repo is read-only except for automation.   I've adjusted the permissions in this repo so that hopefully this is less likely to happen again.  

I've also removed this commit as it's breaking automation and re-landed it at https://github.com/solana-labs/solana/commit/337da184f34855bf362d2409157dfff7fde46890.   Automation should now be unblocked and we'll see it show up in this repo shortly

File: src/connection.js (hunk 1 above)
Comment (Ashry00) [reply src/connection.js:1470]: > 5b4aa213

Menaa

**Pull request merged**